### DISCUSSION TITLE
Server list not correctly updated

### DISCUSSION THREAD

**Discussion by Henri75** [apohl79/audiogridder#1010] [Help]
This small isssue is there for a while now, but checking the log for something else I  decided to go with it.

In the plugin, the server list is not correctly updated : sometimes a server is only with its name, sometimes with its IP. Worst, it change from session to session, so you never now. 

Here is an example where there is only 3 servers among the 6 remote server I use  : one is listed with its name, the other with the IP (see screenshot)
![Screen Shot 2022-09-04 at 16 56 04](https://user-images.githubusercontent.com/56123216/188319886-ccbde6a7-1153-4b35-9865-3ff2a14aa857.png)

In the Tray log I can see the line showing both servers discovered with their IP and name, so it should be reflected in the server list in the tray.

[16:42:37.698|mdns|12b82aff0]   Server(name=Mac-mini, host=192.168.1.31, id=0, uuid=5bfab070-62ea-4152-9762-5a86466594eb, localmode=0, version=dev-build, load=1.87074)
[16:42:37.698|mdns|12b82aff0]   Server(name=Mini, host=192.168.1.10, id=0, uuid=49068b62-971d-4434-b0af-3166f2debb33, localmode=0, version=dev-build, load=8.11236)

Moreover there should be only 3 servers showing in the list (3 are powered on).
127.0.0.1 is not launched (for a couple of weeks) and the 192.168.1.14 too. 
But they still appears in the list, which should'nt  be the case.

**Comment by Henri75** [comment]
I forgot to add, when you manually add a server because it doesnt appear, then at some point AG plugin will still discover it and it will be in double !

See attached screenshot where thinkpad is now in double
![Screen Shot 2022-09-04 at 17 15 32](https://user-images.githubusercontent.com/56123216/188320813-8511ec50-6ed8-4df2-9854-3556acd6d5ec.png)

**Comment by apohl79** [comment]
Yes I hear you. The server management code seems to need some overhaul. Added a feature request for this as well. I think this should probably go into 1.2.

**Comment by Henri75** [comment]
Great ! Thank you again for your amazing work !

Le lun. 5 sept. 2022 à 09:19, Andreas Pohl ***@***.***> a
écrit :
